src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -2616,10 +2616,10 @@ public struct ParamMarshal
                 var paramType = param.Type;
 
                 Class @class;
-                if ( (paramType.GetFinalPointee() ?? paramType).Desugar().TryGetClass(out @class)
-                      && @class.IsRefType)
+                if ( (paramType.GetFinalPointee() ?? paramType).Desugar().TryGetClass(out @class))
                 {
-                    WriteLine("{0} = new {1}();", param.Name, paramType);
+                    if(@class.IsRefType || @class.IsValueType)
```

Comment:
A missing space after the 'if'. Also, doesn't this check actually cover all cases and is therefore redundant? If it's not, wouldn't !@class.IsInterface be better?